OpenTranslator/Controllers/Awesome/PascalTranslationController.cs

```diff
@@ -370,8 +371,8 @@ void InsterRecord(string textId, string originalText, string translatedText, str
                 }
 
                 //or only save the new translation and new translation log
-                ITranslation.Save(translation);
-                ITranslation_Log.Save(translation_log);
+                //ITranslation.Save(translation);
+                //ITranslation_Log.Save(translation_log);
 
```

Comment:
How are you saving the translation object for an existing Text object without this lines? Please take a look at the validation in previous lines:
if this is a translation for a new text (Text entity doesn't exist yet), the if (!ITranslation.IsTextIdAlreadyStoraged(input)) conditional then is **true**, and save the Text entity and its translation (when you are in BlockType.Id block) and then it returns (the following code wont run)
and when the text entity already exists, just save a new translation for it (the lines you commented)

Reply:
I have commented this lines because it inserting record twice in table translation.
when I comment this line then record saved only one time and when I remove comment then saves record twice.